### PR TITLE
adding `nft_info`, `nft_history` and `nfts_by_issuer` clio API models

### PR DESCRIPTION
**Pull request by shawnxie999**
## High Level Overview of Change
Title says all

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: xrpl/models/requests/request.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ class RequestMethod(str, Enum):
     # NFT methods
     NFT_BUY_OFFERS = "nft_buy_offers"
     NFT_SELL_OFFERS = "nft_sell_offers"
+    NFT_INFO = "nft_info"
```
Comment (ckniffen) [inline xrpl/models/requests/request.py:61]: Should this be documented as clio only methods?

**Comment by ckniffen** [comment]
I am unfamiliar with the request api.  Do we first perform a `server_info` call?  If so we should be able to know if it is a clio connection?

We also might want to add a catch for these specific requests that has a nice message if rippled is called and returns `unknownCmd`.

**Comment by shawnxie999** [comment]
> I am unfamiliar with the request api. Do we first perform a `server_info` call? If so we should be able to know if it is a clio connection?
> 
> We also might want to add a catch for these specific requests that has a nice message if rippled is called and returns `unknownCmd`.

@ckniffen No I don't think that's the case. `server_info` is never called implicitly when I tried to set breakpoints in the library code.

**Comment by shawnxie999** [comment]
But I definitely think it's worth to add that in the future, so that we can differentiate between rippled vs clio-only APIs and return some kind of message if they is using the wrong server. Currently it could be pretty confusing to naive users.

File: xrpl/models/requests/request.py (hunk 1 above)
Comment (shawnxie999) [reply xrpl/models/requests/request.py]: added comments

**Review by ledhed2222 — APPROVED** [review]

**Comment by mvadari** [comment]
Has `nfts_by_issuer` been released in a Clio version yet? I'd rather not include it in the libraries before it's been released.

**Comment by shawnxie999** [comment]
> Has `nfts_by_issuer` been released in a Clio version yet? I'd rather not include it in the libraries before it's been released.

Not yet, this API is still in code review

**Comment by mvadari** [comment]
> > Has `nfts_by_issuer` been released in a Clio version yet? I'd rather not include it in the libraries before it's been released.
> 
> Not yet, this API is still in code review

If it can't be used yet, and it's not 100% finalized (anything can happen in code review), it shouldn't be in the libraries.

File: xrpl/models/requests/nft_history.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+"""
+The `nft_history` method retreives  a list of transactions that involved the
```
Comment (ledhed2222) [inline xrpl/models/requests/nft_history.py:2]: accidental extra space before "a"

File: xrpl/models/requests/nft_history.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+from xrpl.models.requests.request import Request, RequestMethod
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class NFTHistory(Request):
+    """
+    The `nft_history` method retreives  a list of transactions that involved the
```
Comment (ledhed2222) [inline xrpl/models/requests/nft_history.py:17]: ditto

**Comment by shawnxie999** [comment]
close this for now since nfts_by_issuer is not supported yet. open a new PR for the existing two APIs https://github.com/XRPLF/xrpl-py/pull/573

**Pull request closed without merging**